### PR TITLE
weechat: build with python3 by default

### PR DESCRIPTION
**Pull request by Ma27**
###### Motivation for this change

WeeChat also supports Python3 for scripts which should be preferred as
CPython2 is about to get EOLed soon: https://weechat.org/scripts/python3/

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -21205,11 +21205,14 @@ in
     inherit (darwin) libobjc;
     inherit (darwin) libresolv;
     guile = guile_2_0;
+    pythonPackages = python3Packages;
   };
 
   weechat = wrapWeechat weechat-unwrapped { };
 
-  weechatScripts = dontRecurseIntoAttrs (callPackage ../applications/networking/irc/weechat/scripts { });
+  weechatScripts = dontRecurseIntoAttrs (callPackage ../applications/networking/irc/weechat/scripts {
```
Comment (jonringer) [inline pkgs/top-level/all-packages.nix]: dontRecurseIntoAttrs is now the default, I think the implementation now is just the id function `drv: drv`
```suggestion
  weechatScripts =  callPackage ../applications/networking/irc/weechat/scripts {
```

Comment (jonringer) [reply pkgs/top-level/all-packages.nix]: yep, had to check https://github.com/NixOS/nixpkgs/blob/5e86891baa57ba4e8aace81c0f24399df3082d37/pkgs/top-level/all-packages.nix#L75

Comment (Ma27) [reply pkgs/top-level/all-packages.nix]: In fact it's always been the default, `dontRecurseIntoAttrs` mainly serves a documentation purpose. To quote the comment above the function:

```
  # This is intended to be the reverse of recurseIntoAttrs, as it is
  # defined now it exists mainly for documentation purposes, but you
  # can also override this with recurseIntoAttrs to recurseInto all
  # the Attrs which is useful for testing massive changes. Ideally,
  # every package subset not marked with recurseIntoAttrs should be
  # marked with this.
```

**Comment by Mic92** [comment]
I agree we should have that at some point. The only thing that concerns me a bit that even archlinux, who were early python3 adopters, build weechat with python2: https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/weechat#n13
On the other hand it seems that most of the python2-only scripts seems quite old: https://weechat.org/scripts/stable/language/python2-only/
Personally I don't use any of plugins, but would be interested how other weechat users think about this.

**Comment by Ma27** [comment]
> I agree we should have that at some point. The only thing that concerns me a bit that even archlinux, who were early python3 adopters, build weechat with python2

Last time I checked (about a month ago) they were building a lot of applications with python2, but I'm no arch user, so could've been a false impression.

> On the other hand it seems that most of the python2-only scripts seems quite old: https://weechat.org/scripts/stable/language/python2-only/

After taking a look at the list, I didn't find any of our currently supported plugins in there. However I guess I should check all of the python stuff manually (just checked `wee-slack` and looked if any of our plugins require python2).

**Comment by Ma27** [comment]
So, short update: rechecked all scripts and it seems as `weechat-xmpp` requires a `xmpppy` with Python3 support, currently investigating if that's possible.

After that we'd need somebody with more darwin experience to fix the build, after that this should be ready IMHO.

**Comment by Ma27** [comment]
Rebased onto current master, also dropped weechat-xmpp as it relies on pretty old dependencies and appears rather unmaintained atm.

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (jonringer) [reply pkgs/top-level/all-packages.nix]: ah, if it has a purpose, then I'm fine with leaving it in :)

**Comment by jonringer** [comment]
There's also a reference to the weechat-xmpp in aliases
```
attribute 'weechat-xmpp' missing, at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/eval-2-shlevy.ewr1.nix.ci/pkgs/top-level/aliases.nix:383:18
```

**Comment by Ma27** [comment]
Ahh, right, fixed! :)

File: pkgs/top-level/aliases.nix [hunk 2]
```diff
@@ -380,7 +380,6 @@ mapAliases ({
   virtviewer = virt-viewer; # added 2015-12-24
   vorbisTools = vorbis-tools; # added 2016-01-26
   webkit = webkitgtk; # added 2019-03-05
-  weechat-xmpp = weechatScripts.weechat-xmpp; # added 2018-09-06
```
Comment (Mic92) [inline pkgs/top-level/aliases.nix]: Maybe?

```
weechat-xmpp = throw "weechat-xmpp has been removed as it doesn't receive
     any updates from upstream and depends on outdated Python2-based modules.";
```

Comment (lheckemann) [reply pkgs/top-level/aliases.nix]: Not in favour. Removal already throws an error, and the reason for removal is documented in the release notes.

**Pull request merged**